docs/blocks/precise-rotation.md

```diff
@@ -488,14 +509,50 @@ Our block JSON file after the above steps should look similar to the code below:
 
 </Spoiler>
 
+<Spoiler title="Shell Example Script">
+
+<CodeHeader>BP/scripts/shell.js</CodeHeader>
+
+```js
+import { world } from "@minecraft/server";
+
+/** @param {number} playerYRotation */
+function getPreciseRotation(playerYRotation) {
+  if (playerYRotation < 0) playerYRotation += 360;
+  const rotation = Math.round(playerYRotation / 22.5);
+
+  return rotation !== 16 ? rotation : 0;
+}
+
+world.beforeEvents.worldInitialize.subscribe(({ blockTypeRegistry }) => {
+  blockTypeRegistry.registerCustomComponent("wiki:shell_rotation", {
+    beforeOnPlayerPlace(e) {
```

Comment:
Would much prefer if `e` was renamed to `eventData` or similar to describe what information it actually contains